### ISSUE TITLE
False negative: Variable is already declare in the upper scope

### ISSUE THREAD

**Issue by numToStr** [eslint/eslint#13644]
**Tell us about your environment**

* **ESLint Version:** 7.8.0
* **Node Version:** 12.18.2
* **npm Version:** 6.14.5

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "env": {
        "es6": true,
        "node": true
    },
    "extends": [
        "eslint:recommended",
        "airbnb-base",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/recommended-requiring-type-checking",
        "prettier",
        "prettier/@typescript-eslint",
        "plugin:node/recommended",
        "plugin:import/errors",
        "plugin:import/warnings",
        "plugin:import/typescript"
    ],
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 2020,
        "sourceType": "module",
        "project": "tsconfig.json"
    },
    "plugins": ["@typescript-eslint"],
    "rules": {
        "max-classes-per-file": ["error", 10],
        "newline-before-return": "error",
        "class-methods-use-this": "off",
        "no-underscore-dangle": ["error", { "allow": ["_id"] }],
        "no-useless-constructor": "off",
        "@typescript-eslint/no-empty-function": [
            "error",
            { "allow": ["constructors"] }
        ],
        "import/no-cycle": "off",
        "import/prefer-default-export": "off",
        "import/extensions": ["error", "never", { "json": "always" }],
        "import/no-default-export": ["error"],
        "@typescript-eslint/unbound-method": "off",
        "@typescript-eslint/no-useless-constructor": "off",
        "@typescript-eslint/explicit-function-return-type": [
            "off",
            { "allowExpression": true }
        ],
        "node/prefer-global/buffer": ["error", "always"],
        "node/prefer-global/console": ["error", "always"],
        "node/prefer-global/process": ["error", "always"],
        "node/prefer-global/url-search-params": ["error", "always"],
        "node/prefer-global/url": ["error", "always"],
        "node/no-deprecated-api": "error",
        "node/no-extraneous-require": "error",
        "node/no-missing-require": "error",
        "node/no-missing-import": [
            "error",
            {
                "tryExtensions": [".js", ".json", ".node", ".ts"]
            }
        ],
        "node/no-unpublished-bin": "error",
        "node/no-unpublished-require": "error",
        "node/no-unsupported-features/es-builtins": "error",
        "node/no-unsupported-features/es-syntax": "off",
        "node/no-unsupported-features/node-builtins": "error",
        "node/process-exit-as-throw": "error",
        "node/shebang": 0
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
export enum AuthType {
    LOCAL = "LOCAL",
    GOOGLE = "GOOGLE",
    FACEBOOK = "FACEBOOK",
}
```

**What did you expect to happen?**
No diagnostics error.

**What actually happened? Please include the actual, raw output from ESLint.**
I am getting `<something> is already declared in the upper scope.`, even if there is nothing in the file. It is weird that this bug is only effecting `enum` declarations.

**Screenshot**
![image](https://user-images.githubusercontent.com/24727447/91852220-23dde300-ec7e-11ea-8fc6-d83315d33a1c.png)

**Comment by anikethsaha** [comment]
try switching off the eslint's `no-shadow` and turning on `typescript-eslint`' s`no-shadow`. 

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/all.ts#L81-L82

**Comment by mdjermanovic** [comment]
Hi @numToStr, thanks for the issue!

You should probably use the following rule, instead of the core `no-shadow` rule: [@typescript-eslint/no-shadow](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md)

**Comment by numToStr** [comment]
@anikethsaha @mdjermanovic Thanks for your feedback. Adding these fixed the issue.
```json
{
  "no-shadow": "off",
  "@typescript-eslint/no-shadow": ["error"]
}
```

Just curious, Why it was not reporting the issue before?

**Comment by mdjermanovic** [comment]
Thanks for the info! I'm closing this since the problem is solved.

> Just curious, Why it was not reporting the issue before?

I'd guess you upgraded `@typescript-eslint` packages to v4, and this is happening now due to some changes in `@typescript-eslint/scope-manager`.

There are probably other variable-related rules that should be replaced with their [typescript plugin extensions](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules). In general, if a rule is on this list, it's probably better to use that rule instead of the equivalent core rule, for linting typescript code.
